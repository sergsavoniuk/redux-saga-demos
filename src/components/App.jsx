import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { push } from 'connected-react-router';

import Routes from 'components/Routes';
import Layout from 'components/Layout';
import AuthForm from 'components/AuthForm';
import Loader from 'components/Loader';
import { login, getIsAuthenticated } from 'redux/auth';

function getUsername() {
  return localStorage.getItem('username');
}

function App({ isAuthenticated, login, push }) {
  useEffect(() => {
    const username = getUsername();
    if (!isAuthenticated) {
      push('/auth');
      if (username != null) {
        login(username);
      }
    }
  }, [isAuthenticated]);

  let Component;
  if (!isAuthenticated) {
    Component = getUsername() != null ? <Loader /> : <AuthForm />;
  } else {
    Component = <Routes />;
  }

  return <Layout>{Component}</Layout>;
}

function mapStateToProps(state) {
  return {
    isAuthenticated: getIsAuthenticated(state),
  };
}

export default connect(
  mapStateToProps,
  { login, push },
)(App);
