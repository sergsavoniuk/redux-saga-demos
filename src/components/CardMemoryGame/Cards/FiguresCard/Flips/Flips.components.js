export { Wrapper, Row, Span } from '../BestTime/BestTime.components';
