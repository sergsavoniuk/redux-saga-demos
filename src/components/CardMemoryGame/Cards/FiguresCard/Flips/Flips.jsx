import React from 'react';

import { Wrapper, Row, Span as Label, Span as Value } from './Flips.components';

function Flips({ matched, wrong }) {
  return (
    <Wrapper>
      <Row>
        <Label>Total Flips:</Label>
        <Value>{2 * (matched + wrong)}</Value>
      </Row>
      <Row>
        <Label>Matched Flips:</Label>
        <Value>{matched}</Value>
      </Row>
      <Row>
        <Label>Wrong Flips:</Label>
        <Value>{wrong}</Value>
      </Row>
    </Wrapper>
  );
}

export default Flips;
