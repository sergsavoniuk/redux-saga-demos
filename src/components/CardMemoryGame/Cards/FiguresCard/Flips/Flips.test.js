import React from 'react';
import { shallow } from 'enzyme';

import Flips from './Flips';
import { Row } from './Flips.components';

describe('<Flips />', () => {
  test('renders correctly without props - uses default values', () => {
    const wrapper = shallow(<Flips />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);

    wrapper.find(Row).forEach(node => {
      expect(
        node
          .children()
          .last()
          .text(),
      ).toMatch('0');
    });
  });

  test('renders correctly with passed props', () => {
    const props = {
      matched: 8,
      wrong: 12,
    };

    const wrapper = shallow(<Flips {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Row)).toHaveLength(3);
  });
});
