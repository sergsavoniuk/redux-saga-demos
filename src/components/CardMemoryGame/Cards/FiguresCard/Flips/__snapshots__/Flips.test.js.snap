// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flips /> renders correctly with passed props 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      Total Flips:
    </styled.span>
    <styled.span>
      40
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Matched Flips:
    </styled.span>
    <styled.span>
      8
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Wrong Flips:
    </styled.span>
    <styled.span>
      12
    </styled.span>
  </styled.div>
</styled.div>
`;

exports[`<Flips /> renders correctly without props - uses default values 1`] = `
<styled.div>
  <styled.div>
    <styled.span>
      Total Flips:
    </styled.span>
    <styled.span>
      0
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Matched Flips:
    </styled.span>
    <styled.span>
      0
    </styled.span>
  </styled.div>
  <styled.div>
    <styled.span>
      Wrong Flips:
    </styled.span>
    <styled.span>
      0
    </styled.span>
  </styled.div>
</styled.div>
`;
